src/index.ts

```diff
@@ -0,0 +1,182 @@
+import express from 'express';
+import { v4 as uuidv4 } from 'uuid';
+import User from './types/User';
+import axios from "axios";
+
+const app = express();
+const port = 3000;
+
+app.use(express.json());
+
+// Initialize the users array with some initial data
+const users: User[] = [
+  { id: 1, login: 'John', password: 'pass123', age: 25, isDeleted: false },
+  { id: 2, login: 'Jane', password: 'pass456', age: 30, isDeleted: false },
+  { id: 3, login: 'Bob', password: 'pass789', age: 35, isDeleted: false }
+];
+
+
+// Create a new user
+app.post('/users', (req, res) => {
+  const { login, password, age } = req.body;
+
+  // Check that all required fields are present in the request body
+  if (!login || !password || !age) {
+    return res.status(400).send({ error: 'Missing required fields' });
+  }
+
+  // Check that the login is valid
+  if (!/^[a-zA-Z0-9]+$/.test(login)) {
+    return res.status(400).send({ error: 'Invalid login' });
+  }
+
+  // Check that the password contains letters and numbers
+  if (!/(?=.*[a-zA-Z])(?=.*\d)/.test(password)) {
+    return res.status(400).send({ error: 'Password must contain letters and numbers' });
+  }
+
+  // Check that the age is between 4 and 130
+  if (age < 4 || age > 130) {
+    return res.status(400).send({ error: 'Invalid age' });
+  }
+
+  // Generate a new ID for the user
+  const id = users.length + 1;
+  const uuid = uuidv4();
+
+  // Create a new user object
+  const user = { id, uuid, login, password, age, isDeleted: false };
+
+  // Add the new user to the users array
+  users.push(user);
+
+  // Return the new user object as the response
+  res.send(user);
+});
+
+// Get user by id
+app.get('/users/:id', (req, res) => {
+  const { id } = req.params;
+  const user = users.find(u => u.id === parseInt(id));
+  if (user) {
+    res.send(user);
+  } else {
+    res.status(404).send({ error: 'User not found' });
+  }
```

Comment:
You could also use the early return pattern here as well, some benefits can be reviewed on this post https://codeburst.io/code-js-in-stlye-early-return-3462323e30d1